libs/vkd3d/device.c

```diff
@@ -5161,6 +5161,18 @@ static HRESULT d3d12_device_init(struct d3d12_device *device,
     return hr;
 }
 
+bool d3d12_device_validate_shader_meta(struct d3d12_device *device, const struct vkd3d_shader_meta *meta)
+{
+    /* TODO: Add more as required. */
+    if (meta->uses_native_16bit_operations && !device->d3d12_caps.options4.Native16BitShaderOpsSupported)
+    {
+        WARN("Attempting to use 16-bit operations in shader %016"PRIx64", but this is not supported.", meta->hash);
```

Comment:
How often does this end up spamming in games we've seen so far?

Reply:
It never happens in any known game.